targets/TARGET_NUVOTON/TARGET_M2351/sleep.c

```diff
@@ -38,9 +38,7 @@ void hal_sleep(void)
     CLK_Idle();
     SYS_LockReg();
 #else
-    SYS_UnlockReg_S();
-    CLK_Idle_S();
-    SYS_LockReg_S();
+    nu_idle_s();
```

Comment:
how is this related to TFM support? Shouldn't this be separate PR addressing sleep issue?

Reply:
Seeing other HAL files, this is related to the TFM ?

Reply:
With TF-M integrated in, secure call overhead gets higher. Combine 3 secure calls into one for TFM build and also for non-TFM build.